### ISSUE TITLE
Is it possible to get the header of a file ?

### ISSUE THREAD

**Issue by d-theo** [adaltas/node-csv-parse#261]
I use the streaming feature of your lib as follow :
```
	csvDownloadStream // download a huge csv, possibly zipped
		.pipe(pipeIf(isGzfile, unGzip))
		.pipe(parser) // csv-parse stream
		.pipe(transformer) // custom transform stream
		.pipe(upStream); // upload the result somewhere else
```

the problem is that I need to upload some metadata of the stream somewhere else like the header of the file or the number of lines. Is it possible to get the header from the `parser` or `parser.info` object  ?

**Comment by wdavidw** [comment]
From the parser instance, you can get a lot of information with the `options`, `state` and `info` objects and you could plug this into one of the available events such as `end`. The `cast` function is also called with a context object but that would be called for every record which is probably not what you wish.

**Comment by d-theo** [comment]
I still cannot find the header, I'm not sure it is stored in some context

**Comment by andrewryan1906** [comment]
@wdavidw I don't understand your answer. Are you saying the options object that is passed into the parser is mutated to add the columns? Because I'm not seeing that.

If not ... how would we be able to obtain the columns during the readable event processing?

**Comment by wdavidw** [comment]
@d-theo did you get answers to your question, can we close ?

**Comment by andrewryan1906** [comment]
Yes thanks
